src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/SaveGCode.java

```diff
@@ -56,17 +59,38 @@ private String addExtension(String name, String [] extensions) {
 		return name + "." + extensions[0];
 	}
 	
+	boolean detailGCodeHeader = true;
+	
 	private void save(String filename, Turtle turtle, Plotter robot) throws Exception {
 		logger.debug("saving...");
 		
 		try (Writer out = new OutputStreamWriter(new FileOutputStream(filename))) {
+			
+			if (detailGCodeHeader) {
+				out.write(";FLAVOR:Marlin-polargraph\n");
+
+				//out.write(";TIME:"+ "\n"); // TODO
+				
+				Rectangle2D.Double bounds = turtle.getBounds();
```

Comment:
it is an excellent idea but it is not related to the title of the PR. One PR = one subject and only one.

Reply:
So a new PR to come :)

Reply:
👍